native-sql-engine/core/src/main/scala/com/intel/oap/execution/ArrowColumnarToRowExec.scala

```diff
@@ -104,25 +125,39 @@ class ArrowColumnarToRowExec(child: SparkPlan) extends ColumnarToRowExec(child =
 
           val beforeConvert = System.nanoTime()
 
-          val instanceID = jniWrapper.nativeConvertColumnarToRow(
-            arrowSchema, batch.numRows, bufAddrs.toArray, bufSizes.toArray,
-            SparkMemoryUtils.contextMemoryPool().getNativeInstanceId)
+          val size = estimateBufferSize(batch.numCols(), batch.numRows()) + totalVariableSize.toInt
+          val directBuffer = ByteBuffer.allocateDirect(size).asInstanceOf[DirectBuffer]
```

Comment:
should alloc memory from the unified mem pool 
https://github.com/oap-project/gazelle_plugin/blob/master/arrow-data-source/common/src/main/scala/org/apache/spark/sql/execution/datasources/v2/arrow/SparkMemoryUtils.scala

Reply:
Updated.